gobblin-config-management/src/main/java/gobblin/config/configstore/ConfigStoreWithResolution.java

```diff
@@ -0,0 +1,33 @@
+package gobblin.config.configstore;
+
+import java.net.URI;
+import java.util.Collection;
+
+import com.typesafe.config.Config;
+
+
+/**
+ * ConfigStoreWithResolution is used to indicate the {@ConfigStore} which support the configuration
+ * resolution by following the imported path.
+ * @author mitu
+ *
+ */
+public interface ConfigStoreWithResolution extends ConfigStore {
+
+  /**
+   * @param uri - the uri relative to this configuration store
+   * @param version - specify the configuration version in the configuration store.
+   * @return - the directly and indirectly specified configuration in com.typesafe.config.Config format for input uri 
+   *  against input configuration version
+   */
+  public Config getResolvedConfig(URI uri, String version);
```

Comment:
We need another method Map<URI, Config> getResolvedConfigs(Collection<URI>, String version) for batched fetches.